### PR TITLE
Fix wrong content id used in building cache key

### PR DESCRIPTION
**Pull request by yurabakhtin**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included
- [x] Changelog was modified

**Other information:**
Issue: https://github.com/humhub/humhub/issues/4857
Related issue: https://github.com/humhub/humhub/issues/4869

### CODE REVIEW

**Review by luke- — CHANGES_REQUESTED** [review]
Unfortunately the ContentOwner interface doesn't provide an ID attribute, what do you think about something like this:

```php
        $result = 'content_' . $content->content->id;
        if (property_exists($content, 'id')) {
            $cacheKey .= '_' . $content->id;
        }
``` 

A better approach would of course be:
- Pass `id` somehow as prefix to the method
- Add getId to `ContentOwner` (Probably not possible yet in Beta due to dependencies & modules etc.)

**Comment by yurabakhtin** [comment]
@luke- Yes, I thought about the solution like `isset($content->id)` or `!empty($content->id)` too, so I agree your solution.

**Comment by yurabakhtin** [comment]
@luke- https://github.com/humhub/humhub/pull/4882/commits/a29e95d8d3866ccfad4af38241f7645ada0f3e90 - I did to use `isset($content->id)` because the `property_exists($content, 'id')` returns `false`, but I didn't use `!empty($content->id)` in order to append the suffix for `0` value even if it may be not happen.

File: protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php [hunk 1]
```diff
@@ -169,7 +169,10 @@ public static function process($text, $options = [])
      */
     public static function buildCacheKeyForContent(ContentOwner $content, $prefix = null)
     {
-        $result = 'content_'.$content->content->id;
+        $result = 'content_' . $content->content->id;
+        if (isset($content->id)) {
+            $result .= '_' . $content->id;
```
Comment (buddh4) [inline protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php:174]: We always need the id of the content and not the record here. Otherwise a wiki page with id 1 will return the same result as post with id 1. The `ContentOwner` interface should always return a `Content` by `$content->content` unfortunately the interface is broken with content active records since it returns an activequery instead of the content itself when calling `getContent()`

File: protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php [hunk 2]
```diff
@@ -284,7 +287,7 @@ protected function renderAbsy($blocks)
             });
         }
 
-        return parent::renderAbsy($blocks);
+        return '';
```
Comment (buddh4) [inline protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php:290]: Why this change this will should break the whole parser process or am I missing something?

**Review by buddh4 — CHANGES_REQUESTED** [review]
So this seems to be an issue with comments? Maybe just add a `instanceof ContentAddon` or something?

File: protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php (hunk 1 above)
Comment (buddh4) [reply protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php:174]: Note there is also a `buildCacheKeyForRecord` which will include the record class.

File: protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php (hunk 2 above)
Comment (yurabakhtin) [reply protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php:290]: I changed this because the `parent::renderAbsy($blocks)` is abstract method and IDE displays error, please review my second screenshot from this comment https://github.com/humhub/humhub/issues/4857#issuecomment-779682809.

Comment (buddh4) [reply protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php:290]: See https://github.com/humhub/humhub/issues/4857#issuecomment-779821434 this seems to be a problem with phpstorm. The method is overwritten in a parent class already.

**Comment by buddh4** [comment]
Also related to https://github.com/humhub/humhub/pull/4870/files I assume that the issue is not the `buildCacheForContent`, but rather that the `ContentOwner` interface is violated in such cases.

File: protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php (hunk 1 above)
Comment (buddh4) [reply protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php:174]: @yurabakhtin I think the following would solve the issue:

```
if($content instanceof ContentAddonActiveRecord) {
    return static::buildCacheKeyForRecord($content);
}

// revert the rest back to prior this PR
```

File: protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php (hunk 2 above)
Comment (yurabakhtin) [reply protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php]: ok, I reverted that here https://github.com/humhub/humhub/pull/4882/commits/27e2022aebcf6c84b8098b715a9566dc47360075

File: protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php (hunk 1 above)
Comment (yurabakhtin) [reply protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php]: @buddh4 Yes, your solution works well, so I applied this here https://github.com/humhub/humhub/pull/4882/commits/7c31388d80703f3643b4e11fc81fd32862162532 and also reverted there the fix from PR https://github.com/humhub/humhub/pull/4870/files

**Review by buddh4 — APPROVED** [review]

File: protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php (hunk 2 above)
Comment (ArchBlood) [reply protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php]: Just an idea and could be ignored here if a better method could be implemented

```php
return parent::renderAbsy($blocks['content']);
```
or
```php
return rtrim(parent::renderAbsy($blocks['content']));
```

Comment (yurabakhtin) [reply protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php]: @buddh4 Do you agree to apply `rtrim()` there?

Comment (buddh4) [reply protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php]: No, please leave as is.
